Update how to contact us.rst (#8108)
mattermost/docs: source/deploy/server/orchestration.rst

This is an optional program for third-party developers to increase awareness about their work and to enable Mattermost to refer its communities to different solutions.

> Individuals or companies interested in exploring deployment solutions with Mattermost are encouraged to reach out to us via the Mattermost Partner Form. Simply select "Deployment Solutions" under the "Business Type" section. You can access the form here: [Mattermost Partner Program](https://support.mattermost.com/hc/en-us/requests/new?ticket_form_id=24278352368532)

Reviewer: ```suggestion
Individuals or companies interested in exploring deployment solutions with Mattermost are encouraged to reach out to us via the `Mattermost Partner Program Form <https://support.mattermost.com/hc/en-us/requests/new?ticket_form_id=24278352368532>`_. On the form, select **Deployment Solutions** under the **Business Type** section to get started.
```
